src/Http/samples/HttpApiSampleApp/Startup.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+namespace HttpApiSampleApp
+{
+    public class Startup
+    {
+        // This method gets called by the runtime. Use this method to add services to the container.
+        // For more information on how to configure your application, visit https://go.microsoft.com/fwlink/?LinkID=398940
+        public void ConfigureServices(IServiceCollection services)
+        {
+        }
+
+        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+        {
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+            }
+
+            app.UseRouting();
+
+            app.UseEndpoints(endpoints =>
+            {
+                [HttpPost("/EchoTodo")]
+                JsonResult EchoTodo([FromBody] Todo todo) => new(todo);
+
+                endpoints.MapAction((Func<Todo, JsonResult>)EchoTodo);
```

Comment:
The cast is a bit weird here - do any of the language enhancements we've discussed help out with this kind of methodgroup-to-delegate conversion?

Reply:
Yep

Reply:
cc @jaredpar @MadsTorgersen

Reply:
Brain is a bit fried today but not sure which enhancement we discussed that would fix this. 

If this were written as a lambda I agree the discussions we've had would help. It would have a natural type, we'd synthesize a `delegate`, use that and it would become the type passed to `MapAction(Delegate)`.  This is a method group though, not a lambda. Don't recall us talking about enhancing those. 

It's not outside the realm of possibilities, I just haven't thought about it a bunch yet.

Reply:
This came up as well. The idea is that overloads aren't supported we didn't understand why this would be problematic.

Reply:
Brain is feeling a bit better today: medium rare, no longer well done. 

Yeah we did talk about this. It's the same idea: giving method groups a natural type. That lets them have a tie breaker in cases where a `Delegate` overload exists.

Reply:
@jaredpar We should figure out what preview release of .NET 6 we could expect a version of this. We'll likely have something in preview2 and I don't expect we'd have anything that early in the compiler timeframe but we should target a release because this is pretty unusable in C# without it 😄